pkg/apiserver/profiling/service.go

```diff
@@ -87,11 +87,15 @@ var newService = fx.Provide(func(lc fx.Lifecycle, p ServiceParams, fts *fetchers
 	lc.Append(fx.Hook{
 		OnStart: func(ctx context.Context) error {
 			s.lifecycleCtx = ctx
-			s.wg.Add(1)
+			s.wg.Add(2)
 			go func() {
 				defer s.wg.Done()
 				s.serviceLoop(ctx)
 			}()
+			go func() {
+				defer s.wg.Done()
+				s.migrateLegacyState(ctx)
+			}()
```

Comment:
I think we can do the migration work at this moment. How do you think? @breeswish

Reply:
How about move it to L78? As you can see in L78 there is already a "autoMigrate" process, which does similar things.

Reply:
OK, I thought making it run as a goroutine can reduce the start time, but it is not necessary.